src/editor/languages.js

```diff
@@ -27,6 +27,9 @@ export const LANGUAGES = [
     new Language("text", "Plain Text", null, null),
     new Language("math", "Math", null, null),
     new Language("javascript", "JavaScript", javascriptLanguage.parser, "js", true),
+    new Language("typescript", "TypeScript", typescriptLanguage.parser, "ts", true),
+    new Language("jsx", "JSX", jsxLanguage.parser, "jsx", true),
+    new Language("tsx", "TSX", tsxLanguage.parser, "tsx", true),
```

Comment:
I think the fourth argument can be `null` for JSX and TSX, since guesslang doesn't support them.

Reply:
Oh, I thought the last argument represented if the language supports formatting or not?

Reply:
Wait, my bad, I misinterpreted what you said. Gotcha! I'll do that right now.